graph/travelling_salesman_problem.cpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file
+ * @brief [Travelling Salesman Problem]
+ * (https://en.wikipedia.org/wiki/Travelling_salesman_problem)
+ *
+ * @author [Mayank Mamgain](http://github.com/Mayank17M)
+ *
+ * @details
+ * Travelling salesman problem asks:
+ * Given a list of cities and the distances between each pair of cities, what is
+ * the shortest possible route that visits each city exactly once and returns to
+ * the origin city?
+ * TSP can be modeled as an undirected weighted graph, such that cities are the
+ * graph's vertices, paths are the graph's edges, and a path's distance is the
+ * edge's weight. It is a minimization problem starting and finishing at a
+ * specified vertex after having visited each other vertex exactly once.
+ * This is the naive implementation of the problem.
+ */
+
+#include <limits.h>  /// header for limits of integral types
+
+#include <algorithm>  /// for std::min
+#include <cassert>    /// for assert
+#include <iostream>   /// for IO operations
+#include <vector>     /// header for std::vector
+
+/**
+ * @namespace graph
+ * @brief Graph Algorithms
+ */
+namespace graph {
+
+/**
+ * @brief Function calculates the minimum path distance that will cover all the
+ * cities starting from the source.
+ *
+ * @param cities matrix representation of cities
+ * @param src Point from where salesman is starting
+ * @param V number of vertices in the graph
+ *
+ */
+
+int TravellingSalesmanProblem(std::vector<std::vector<int>> *cities, int src,
+                              int V) {
+    //// vtx stores the vertexs of the graph
+    std::vector<int> vtx;
+    for (int i = 0; i < V; i++) {
+        if (i != src) {
+            vtx.push_back(i);
+        }
+    }
+
+    //// store minimum weight Hamiltonian Cycle.
+    int min_path = INT_MAX;
+    do {
+        //// store current Path weight(cost)
+        int curr_weight = 0;
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let me know if you need any help with that**.